types/stripe/index.d.ts

```diff
@@ -5,11 +5,12 @@
 //                 Amrit Kahlon <https://github.com/amritk/>
 //                 Adam Cmiel <https://github.com/adamcmiel>
 //                 Justin Leider <https://github.com/jleider>
+//                 Kamil Gałuszka <https://github.com/galuszkak>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
```

Comment:
Maybe it will be best on example. Let's emulate 3 possible options. 
* Only API v2
* Only API v3
* API v2 and v3 together

If you load only API v2:
`Stripe.applePay` or `Stripe. setPublishableKey` is available for You.
If you load only API v3  those methods are unavailable for You.

If you load APIv2 and APIv3. Methods above are Available on Stripe object.

So if you load APIv2 and APIv3 Stripe object is merging method from v2 and v3. 

I think the best idea now is to use typescript namespace merging ( v2 will have exported methods/properties ) and if someone include `@types/stripe-v2` and `@types/stripe-v3`, will have both methods exposed. If only one of them then only method for specific version available.